### PR TITLE
6.4 hardening 3

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java [hunk 1]
```diff
@@ -636,6 +636,7 @@ public String toString() {
     private boolean required = false;
     private boolean internal = false;
     private Set<String> possibleValues = null;
+    private Integer maximumValues = null;
```
Comment (teryk) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java]: name is confusing. We don't know if it's the max value allowed or the maximum number of value allowed. 
Suggestions: `maximumNumberOfValues`, `maximumValueCount`, `valueCountLimit`, `maxNbValues`, `maxNumberOfValues`.
It's in the API so it's better to get it right

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/CreateAction.java [hunk 2]
```diff
@@ -137,6 +140,7 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
+    ruleWsSupport.checkQProfileAdminPermission();
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/CreateAction.java]: I've talked with Daniel and he will work on it

File: server/sonar-server/src/test/java/org/sonar/server/rule/ws/CreateActionTest.java [hunk 3]
```diff
@@ -59,20 +65,26 @@
...
 
   private TestOrganizationFlags organizationFlags = TestOrganizationFlags.standalone();
+  private DefaultOrganizationProvider defaultOrganizationProvider = TestDefaultOrganizationProvider.from(db);
 
   private WsActionTester wsTester = new WsActionTester(new CreateAction(db.getDbClient(),
     new RuleCreator(system2, new RuleIndexer(es.client(), db.getDbClient()), db.getDbClient(), newFullTypeValidations(),
       TestDefaultOrganizationProvider.from(db)),
-    new RuleMapper(new Languages(), createMacroInterpreter()), organizationFlags));
+    new RuleMapper(new Languages(), createMacroInterpreter()), organizationFlags,
+    new RuleWsSupport(mock(DbClient.class), userSession, defaultOrganizationProvider)));
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/rule/ws/CreateActionTest.java]: Bad copy paster

File: server/sonar-server/src/test/java/org/sonar/server/rule/ws/CreateActionTest.java [hunk 4]
```diff
@@ -169,10 +183,32 @@ public void fail_to_create_rule_when_organizations_are_enabled() throws Exceptio
...
+  public void throw_ForbiddenException_if_not_profile_administrator() throws Exception {
+    userSession.logIn();
+
+    expectedException.expect(ForbiddenException.class);
+
+    wsTester.newRequest().execute();
+  }
+
+  @Test
+  public void throw_UnauthorizedException_if_not_logged_in() throws Exception {
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/rule/ws/CreateActionTest.java]: I haven't changed behaviour, just added a test

File: sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java (hunk 1 above)
Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java]: Indeed, so let's choose maximumValueCount.

**Pull request merged**